kernel/exit.c

```diff
@@ -281,6 +289,11 @@ int do_wait(int idtype, pid_t_ id, struct siginfo_ *info, struct rusage_ *rusage
         info->child.pid = id;
         if (reap_if_needed(task, info, rusage, options))
             goto found_zombie;
+        if (task->ptrace.stopped && task->ptrace.signal) {
+            info->child.status = task->ptrace.trap_event << 16 | task->ptrace.signal << 8 | 0x7f;
+            task->ptrace.signal = 0;
+            goto found_ptrace_stopped;
+        }
```

Comment:
Could this be moved into reap_if_needed to avoid duplication?

Reply:
I feel like there was a reason I put it here but I might have refactored it so that it doesn't matter anymore. I'll give it a try.

Reply:
Err, no, I figured out why. I need to go to `found_ptrace_stopped`. I guess I could put the condition and that code into `reap_if_needed` and return false and then check again but that seemed a bit ugly…

Reply:
Could you put the `info->sig = SIGCHLD_` in some kind of if statement so it doesn't happen if it's not supposed to? Or put it in reap_if_zombie?